env/main.tf

```diff
@@ -357,6 +357,14 @@ resource "azurerm_api_management_api_policy" "patient_policy" {
       <value>@((string)context.Variables["funchostkey"])</value>
     </set-header>
   </inbound>
+  <outbound>
+    <base />
+    <choose>
+      <when condition="@(context.Response.StatusCode == 401)">
+        <cache-remove-value key="func-host-key" />
```

Comment:
Does this require the request to be resent, to get a new key?

Reply:
Yes, or I can just update the cache here.

Reply:
now it will update the host key when 401.